src/geckomat/gather_kcats/runDLKcatFromDocker.m

```diff
@@ -0,0 +1,111 @@
+%                   the default model adapter)
+%   DLKcatPath      path where DLKcat is/will be installed. (Optional,
+%                   defaults to GECKO/dlkcat)
+%
+%   NOTE: 1. Requires Docker to be installed. Visit "https://www.docker.com"
+%         2. Runtime will depend on whether the image is to be downloaded or not.
+
+%Get the GECKO path
+geckoPath = findGECKOroot();
+
+if nargin < 4 || isempty(DLKcatPath)
+    DLKcatPath = fullfile(geckoPath,'dlkcat');
+end
+
+if nargin < 3 || isempty(modelAdapter)
+    modelAdapter = ModelAdapterManager.getDefaultAdapter();
+    if isempty(modelAdapter)
+        error('Either send in a modelAdapter or set the default model adapter in the ModelAdapterManager.')
+    end
+end
+params = modelAdapter.params;
+
+if nargin < 2 || isempty(DLKcatFile)
+    DLKcatFile = fullfile(params.path,'data','DLKcat.tsv');
+end
+
+if nargin < 1 || isempty(DLKcatPath)
+    deleteImage = false;
+end
+
+%% Check and install requirements
+% On Mac, docker might not be properly loaded if MATLAB is started via
+% launcher and not terminal.
+if ismac
+    setenv('PATH', strcat('/usr/local/bin', ':', getenv("PATH")));
+end
+
+% Check if docker is installed
+[checks.docker.status, checks.docker.out] = system('docker --version');
+if checks.docker.status ~= 0
+    error('Cannot find Docker. Make sure it is installed')
+end
+
+% Dowload DLKcat package
+if ~exist(fullfile(DLKcatPath,'DLKcat.py'),'file')
+    if ~exist(fullfile(DLKcatPath),'dir')
+        mkdir(fullfile(DLKcatPath));
+    end
+    disp('=== Downloading DLKcat ...')
+    packageURL = 'https://github.com/SysBioChalmers/GECKO/raw/dlkcatPackage/dlkcat.zip';
+    %packageURL = 'https://github.com/SysBioChalmers/GECKO/releases/download/v3.0.0/dlkcat_package.zip';
+    websave(fullfile(DLKcatPath,'dlkcat_package.zip'),packageURL);
+    unzip(fullfile(DLKcatPath,'dlkcat_package.zip'),geckoPath);
+    delete(fullfile(DLKcatPath,'dlkcat_package.zip'));
+end
+
+currPath = pwd();
+cd(DLKcatPath);
+
+% First check if the Docker image exist
+[checks.image.status, checks.image.out] = system('docker images');
+if checks.image.status == 0 && ~contains(checks.image.out,'dlkcat')
+    % Check if the image file is already dowloaded
+    if ~exist(fullfile(DLKcatPath,'dlkcat_docker.tar.gz'),'file')
+        disp('=== Downloading DLKcat docker image, this may take several minutes...')
+        packageURL = 'https://github.com/SysBioChalmers/GECKO/raw/dlkcatPackage/dlkcat_docker.tar.gz';
+        websave(fullfile(DLKcatPath,'dlkcat_docker.tar.gz'),packageURL);
+    end
+    [checks.dockerload.status, checks.dockerload.out] = system('docker load --input dlkcat_docker.tar.gz');
+    if deleteImage
+        delete(fullfile(DLKcatPath,'dlkcat_docker.tar.gz'));
+    end
+    if checks.dockerload.status ~= 0
+        error('Fail to load the image')
+    end
+elseif checks.image.status ~= 0
+    error('Docker is installed. Check if the docker desktop app is running')
+end
```

Comment:
If the image were to be published, most of this code would not be needed. So that is what I would strongly advise, especially since it would bring transparency to how the image was created.

Reply:
I'm guessing it's a private personal repository. If it's a pure `DLKcat` image, I guess it would make sense to have it in the [`DLKcat` repository](https://github.com/SysBioChalmers/DLKcat/).

Reply:
It contains the minimum code and data to predict kcat values, while the code in DLKcat repo can also regenerate the underlying DL model. Moreover, the `dlkcat.py` script here is modified to use the GECKO input file (it for instance has an extra column with reaction identifiers which are then also included in the output file, and finding SMILES is disabled as it is done in MATLAB instead).

But perhaps relevant to mention is that the `dlkcat.zip` and `DLKcat` image are stored in a separate orphan branch in GECKO GitHub.

Reply:
> `dlkcat.zip` and `DLKcat` image are stored in a separate orphan branch in GECKO GitHub.

I think we ought to do better than that. In terms of the image itself, it should be [stored on GH and produced with an action at every release](https://docs.github.com/en/actions/publishing-packages/publishing-docker-images#publishing-images-to-github-packages). @ae-tafur can I give this a go with your code? And then this branch can get some more commits to simplify the code.

Reply:
I also see DLKcat as more of a plugin to GECKO. That's why I would also not already set the paths when running GECKOInstaller.Install.